substrate/frame/nomination-pools/src/lib.rs

```diff
@@ -2934,6 +3014,13 @@ impl<T: Config> Pallet<T> {
 		T::PalletId::get().into_sub_account_truncating((AccountType::Bonded, id))
 	}
 
+	pub fn migrate_to_delegate_stake(id: PoolId) -> DispatchResult {
+		T::StakeAdapter::migrate_nominator_to_agent(
+			&Self::create_bonded_account(id),
```

Comment:
At this point, I think it would be useful to rename `fn create_bonded_account` to `fn generate_bonded_account_id` or similar. It is already confusing and it gets worse as its used more.

Same for `fn create_reward_acount`.

Perhaps something like this?

```rust
pub fn generate_account_id(id: PoolId, type: AccountType) -> T::AccountId {
  T::PalletId::get().into_sub_account_truncating((type, id))
}
```

Reply:
Refactored 
- `create_bonded_account` --> `generate_bonded_account`
- `create_reward_account` --> `generate_reward_account`